docker_ci_deploy/tests/test_main.py

```diff
@@ -4,67 +4,156 @@
 import sys
 from subprocess import CalledProcessError
 
+import pytest
 from testtools import ExpectedException
 from testtools.assertions import assert_that
 from testtools.matchers import (
-    AfterPreprocessing as After, Equals, MatchesRegex, MatchesStructure, Not)
+    AfterPreprocessing as After, Equals, Is, MatchesRegex, MatchesStructure,
+    Not)
 
 from docker_ci_deploy.__main__ import (
-    cmd, DockerCiDeployRunner, main, strip_image_tag)
+    cmd, DockerCiDeployRunner, join_image_tag, main, replace_image_registry,
+    replace_tag_version, split_image_tag)
 
-""" strip_image_tag() """
+""" split_image_tag() """
```

Comment:
I wanted to but didn't want to make the PR too big. But I guess it's big already...